enhancements/windows-containers/csi-proxy.md

```diff
@@ -0,0 +1,160 @@
+
+### API Extensions
+
+N/A
+
+### Risks and Mitigations
+
+If there are any security issues present in the CSI driver images deployed by the user, they will not be fixed
+by upgrading OCP. It is up to the user to stay informed and update their images as necessary.
+
+### Drawbacks
+
+The main drawback to this approach is it puts a lot of work on the cluster administrator. As they will be required
+to deploy and maintain the Windows CSI DaemonSet themselves, and will not be given support for any issues that may
+come up, which do not specifically deal with the csi-proxy Windows service. This may cause an increased load
+on both support and developers, as support issues coming in will need to be properly filtered out based on the root
+cause of the problem.
+
+## Design Details
+
+### Test Plan
+
+WMCO already has an extensive end to end test suite. To facilitate testing the additions described in this enhancement,
+the e2e tests will be augmented to deploy a CSI node driver DaemonSet across Windows nodes, just as a user would. The
+e2e tests would then validate that a Windows pod is able to mount a volume backed by the driver.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+This feature is targeted for WMCO 9.0.0 in OpenShift 4.14.
+
+#### Removing a deprecated feature
+
+No action required.
+
+### Upgrade / Downgrade Strategy
+
+[When upgrading cluster versions, in-tree driver Volumes will automatically be updated to use the appropriate CSI node
+drivers by the cluster](https://docs.openshift.com/container-platform/4.12/storage/container_storage_interface/persistent-storage-csi-migration.html).
+
+If users are making use of the Linux CSI node drivers on the cluster, and not deploying all parts of storage
+themselves, some extra care must be had when upgrading a cluster.
+Before upgrading, users should check the OCP release notes to determine how the Linux CSI drivers have changed, and
+if they need to use a new Windows CSI node driver image. If so, the Windows DaemonSet should be updated once the
+cluster upgrade is complete.
+
+WMCO does not support downgrades.
+
+### Version Skew Strategy
+
+If the CSI driver pods included in OCP are used as the driver controller and Linux node drivers, while the Windows
+node drivers use the upstream images, there is going to be a mismatch. This should be mitigated by the cluster
+administrator ensuring that the upstream images they use are as close to a version match as possible with the OCP
+distributed drivers.
+
+To eliminate this issue, the cluster administrator can choose to use an OCP cluster with no CSI drivers installed.
+They would then install both the Windows and Linux CSI pods and required resources themselves, using upstream
+documentation. This is only an option if they only plan on using storage on Windows, as doing this will prevent the
+user from getting support from Red Hat for any storage issues they may have, both Linux and Windows.
+
```

Comment:
You should also mention using a separate cluster for Windows container workloads.